prov/sockets/src/sock_progress.c

```diff
@@ -228,9 +235,19 @@ static struct sock_pe_entry *sock_pe_acquire_entry(struct sock_pe *pe)
 	return pe_entry;
 }
 
+static int sock_pe_is_completion_reported(struct sock_pe_entry *pe_entry)
+{
+	int completion_reported = ofi_atomic_inc32(&pe_entry->completion_reported);
+	return (completion_reported != 1);
+}
+
 static void sock_pe_report_send_cq_completion(struct sock_pe_entry *pe_entry)
 {
 	int ret = 0;
+
+	if (sock_pe_is_completion_reported(pe_entry))
+		return;
```

Comment:
Thanks, I've applied the modifications you have suggested. 
I created a new PR (see PR #3221) that extracts the patches that are not directly related to the new connection manager for the sockets provider.
We can continue the discussion here after PR #3221 get merged.